Source/CarthageKit/Cartfile.swift

```diff
@@ -31,24 +31,30 @@ public struct Cartfile {
 		var result: Result<(), CarthageError> = .success(())
 
 		string.enumerateLines { line, stop in
-			let scanner = Scanner(string: line)
+			let scannerWithComments = Scanner(string: line)
 
-			if scanner.scanString(Cartfile.commentIndicator, into: nil) {
+			if scannerWithComments.scanString(Cartfile.commentIndicator, into: nil) {
 				// Skip the rest of the line.
 				return
 			}
 
-			if scanner.isAtEnd {
+			guard !scannerWithComments.isAtEnd,
+				let remainingString = scannerWithComments.remainingSubstring else {
```

Comment:
I think we have to stop parsing and surface an error if `Scanner.remainingSubstring` returns `nil`:

```swift
guard let remainingString = scannerWithComments.remainingSubstring.map(String.init) else {
	result = .failure(CarthageError.internalError(
		description: "Can NSScanner split an extended grapheme cluster? If it does, this will be the error…"
	))
 	stop = true
 	return
}
```

→ also removing the `String(…)` before `. strippingTrailingCartfileComment`